src/Security/Authentication/test/CookieTests.cs

```diff
@@ -151,6 +151,37 @@ public async Task SignInCausesDefaultCookieToBeCreated()
             Assert.Equal("no-cache", transaction.Response.Headers.Pragma.ToString());
         }
 
+        [Fact]
+        public async Task CustomAuthSchemeEncodesCookieName()
```

Comment:
Are spaces the only thing we have issues with?  Since we allow arbitrary strings for scheme names, do we just throw in those cases?

Reply:
Or do those get encoded fine?  I guess should we try more than just spaces in here?

Reply:
Spaces were the specific complaint, but the encoding can handle anything. I'll add some unicode as proof. This is the same encoding we used to use, only applied differently to avoid the decoding problems.